blockchain/blocks/src/tipset.rs

```diff
@@ -133,6 +134,21 @@ impl Tipset {
     pub fn weight(&self) -> &BigInt {
         self.min_ticket_block().weight()
     }
+    /// Returns true if self wins according to the filecoin tie-break rule
+    /// (FIP-0023)
+    pub fn break_weight_tie(&self, other: &Tipset) -> bool {
+        // blocks are already sorted by ticket
+        for (a, b) in self.blocks().iter().zip(other.blocks().iter()) {
+            let ticket = a.ticket().as_ref().unwrap();
```

Comment:
Is it safe to unwrap here? If yes, let's put a comment on why. If not, perhaps we should take care of the error in a more gentle way?

Reply:
Normally yes. Function `block_sanity_checks` should have been called at this point. What you think?

Reply:
In that case, use `expect()`:

```suggestion
            let ticket = a.ticket().as_ref().expect("The function block_sanity_checks() should have been called at this point.");
```